Modelica/Blocks/Logical.mo

```diff
@@ -973,6 +973,41 @@ parameter \"terminationText\".
 
 </html>"));
   end TerminateSimulation;
+
+  block AssertCondition "Trigger error (or warning) and print message if assertion condition is not fulfilled"
+
+    Modelica.Blocks.Interfaces.BooleanOutput condition=true
+      "Trigger error (or warning) and print message when condition becomes false"
+      annotation (Dialog, Placement(transformation(extent={{200,-10},{220,10}})));
```

Comment:
I see that it comes from `TerminateSimulation`, but what is the purpose of the argument-less `Dialog` annotation?

Reply:
Without the **Dialog** annotation, the user is not able to specify **condition** via the parameterization/dialog window (at least in Dymola).

Reply:
Compare `BooleanExpression`, where the `Dialog` isn't argument-less:
```
Modelica.Blocks.Interfaces.BooleanOutput y = false "Value of Boolean output" annotation(Dialog(group = "Time varying output signal"), …);
```

Reply:
Understood regarding the group annotation in BooleanExpression (and RealExpression, etc.), but still I can assure you that Dialog without an argument does have functionality in at least Dymola. It's effectively equivalent to doing `annotation(Dialog(group="Parameters"))` as far as I can tell. Though, I suppose with that in mind it could be argued it doesn't belong under a group called "Parameters" since it's not a parameter.

Reply:
> suggesting to use `annotation(Dialog(group))` rather than `annotation(Dialog)` ...

But not argument-less `group`, of course?!

Reply:
Yes, while waiting for something like `Dialog(show = true)` to become standardized, one needs to make up some other information instead.  As `tab` defaults to `"General"`, perhaps the best alternative is to set `Dialog(tab = "General")`?

Reply:
A problem with `enable` is that it is true by default, so one cannot assume that setting it to that value would have any effect.

Reply:
Block has been restructured to use input signal rather than dialog, so this issue is no longer applicable